resources/lib/videoplayer.py

```diff
@@ -222,10 +222,13 @@ def check_skipping(self):
             skip_time_start = self._stream_data.skip_events_data.get(skip_type).get('start')
             skip_time_end = self._stream_data.skip_events_data.get(skip_type).get('end')
 
-            if skip_time_start <= current_time <= skip_time_end:
-                self._ask_to_skip(skip_type)
-                # remove the skip_type key from the data, so it won't trigger again
-                self._stream_data.skip_events_data.pop(skip_type, None)
+            if skip_time_start <= current_time < skip_time_end:
+                if G.args.addon.getSetting("ask_before_skipping") != "true":
+                    self._instaskip(skip_type)
+                else:
+                    self._ask_to_skip(skip_type)
+                    # remove the skip_type key from the data, so it won't trigger again
+                    self._stream_data.skip_events_data.pop(skip_type, None)
```

Comment:
This one needs to be pulled back by 4 spaces, so it's below `if skip_time_start <= current_time < skip_time_end:` - otherwise it will only be popped when asked to skip, but not when instaskipping.

Otherwise it looks good to me, thank you!